pkg/distributor/ha_tracker_test.go

```diff
@@ -70,6 +70,54 @@ func checkReplicaTimestamp(t *testing.T, duration time.Duration, c *haTracker, u
 	})
 }
 
+func TestHATrackerCacheSyncOnStart(t *testing.T) {
+	const cluster = "c1"
+	const replica = "r1"
+
+	var c *haTracker
+	var err error
+	var now time.Time
+
+	codec := GetReplicaDescCodec()
+	kvStore, closer := consul.NewInMemoryClient(codec, log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	mock := kv.PrefixClient(kvStore, "prefix")
```

Comment:
we don't need the preifxed client, do we?

Reply:
You are right. Fixed it [2e20498](https://github.com/grafana/mimir/pull/9826/commits/2e204988d140eefe7d1459758b85afffc4803882)

Reply:
I am trying to create a mock client and followed the approach from other tests. However might not needed 🤔 . When I remove it though tests are failing. This code might be replaced at the next PR in favor of [this](https://github.com/grafana/mimir/pull/9826#discussion_r1841194320) .

Reply:
Related PR: https://github.com/grafana/mimir/pull/9905